core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -1197,27 +1199,40 @@ wasi_sock_recv(wasm_exec_env_t exec_env, wasi_fd_t sock, iovec_app_t *ri_data,
         || !validate_native_addr(ri_data, (uint32)total_size))
         return __WASI_EINVAL;
 
-    /* recv ri_data one by one */
+    /* receive and scatter*/
+    for (i = 0; i < ri_data_len; i++, ri_data++) {
+        total_size += ri_data->buf_len;
+    }
+    if (total_size >= UINT32_MAX
+        || !(buf_begin = wasm_runtime_malloc((uint32)total_size)))
```

Comment:
Had better add check `total_size > 0`, if ri_data_len is 0, total_size will be 0, we had better not malloc memory, or, in wasm_runtime_malloc, a warning will be prompted:
```bash
warning: wasm_runtime_malloc with size zero
```
```C
    if (total_size >= UINT32_MAX
        || (total_size > 0
            && !(buf_begin = wasm_runtime_malloc((uint32)total_size))))
```